pennylane/grouping/utils.py

```diff
@@ -0,0 +1,492 @@
+        raise ValueError(
+            "Symplectic vector space must have even dimension, instead got vectors of shape {}."
+            .format(np.shape(pauli_vec_1))
+        )
+
+    if not (
+            np.array_equal(pauli_vec_1, pauli_vec_1.astype(bool))
+            and np.array_equal(pauli_vec_2, pauli_vec_2.astype(bool))
+        ):
+        raise ValueError(
+            "Vectors a and b must have strictly binary components, instead got {} and {}".format(
+                pauli_vec_1, pauli_vec_2
+            )
+        )
+
+    n_qubits = int(len(pauli_vec_1) / 2)
+
+    for i in range(n_qubits):
+
+        a_x_i = pauli_vec_1[i]
+        a_z_i = pauli_vec_1[n_qubits + i]
+        b_x_i = pauli_vec_2[i]
+        b_z_i = pauli_vec_2[n_qubits + i]
+
+        if not (
+                ((a_x_i == a_z_i == 0) or (b_x_i == b_z_i == 0))
+                or ((a_x_i == b_x_i) and (a_z_i == b_z_i))
+            ):
+            return False
+
+    return True
+
+
+def convert_observables_to_binary(observables):
+    """Converts a list of Pauli words to the binary vector representation and yields column matrix
+        of the binary vectors.
+
+    The dimension of the binary vectors will be implied from the highest wire being acted on
+    non-trivially by the Pauli words in observables.
+
+    **Usage example:**
+
+    >>> convert_observables_to_binary([PauliX(0) @ PauliY(2), PauliZ(0) @ PauliZ(1) @ PauliZ(2)])
+    array([[1., 0.],
+           [0., 0.],
+           [1., 0.],
+           [0., 1.],
+           [0., 1.],
+           [1., 1.]])
+
+    Args:
+        observables (list[Union[Identity, PauliX, PauliY, PauliZ, Tensor]]):
+
+    Returns:
+        array[bool]: a column matrix of the observables in binary vector representation.
+
+    """
+
+    m_cols = len(observables)
+
+    n_qubits = get_n_qubits(observables)
+
+    binary_mat = np.zeros((2 * n_qubits, m_cols))
+
+    for i in range(m_cols):
+        binary_pauli = pauli_to_binary(observables[i], n_qubits)
+
+        binary_mat[:, i] = binary_pauli
+
+    return binary_mat
+
+
+def get_qwc_compliment_adj_matrix(binary_observables):
+    """Obtains the adjacency matrix for the complementary graph of the qubit-wise commutativity
+    graph for a given set of observables in the binary representation.
+
+    The qubit-wise commutativity graph for a set of Pauli words has a vertice for each Pauli word,
+    and two nodes are connected if and only if the corresponding Pauli words are qubit-wise
+    commuting.
+
+    **Usage example:**
+
+    >>> binary_observables
+    array([[1., 0., 0.],
+           [0., 1., 0.],
+           [1., 1., 0.],
+           [0., 1., 1.],
+           [0., 0., 0.],
+           [1., 1., 0.]])
+    >>> get_qwc_compliment_adj_matrix(binary_observables)
+    array([[0., 1., 1.],
+           [1., 0., 0.],
+           [1., 0., 0.]])
+
+    Args:
+        binary_observables (array[bool]): a column matrix of Pauli words in the binary vector
```

Comment:
array[array[bool]] I think?

Reply:
`binary_observables` should be a binary array. I've changed the description to "a matrix whose columns are the Pauli words in the binary vector representation", to avoid confusion.